src/App/Part.cpp

```diff
@@ -104,6 +105,37 @@ PyObject *Part::getPyObject()
     return Py::new_reference_to(PythonObject);
 }
 
+std::vector<DocumentObject*> Part::addObjects(std::vector<DocumentObject*> objs)
```

Comment:
Part support for VarSets

Reply:
This won't work if the object is accessed in a generic way. For example, when it's checked whether an object has "grouping" behaviour then it's done this way:
```
App::DocumentObject* obj = ...
if (obj->hasExtension(GroupExtension::getExtensionClassTypeId())) {
   App::GroupExtension* grp = obj->getDocumentObjectPtr()->getExtensionByType<GroupExtension>();
   grp->addObjects(...);
...
   grp->removeObjects(...);
}
```
If `obj` is an `App::Part` then the calls of addObjects() or removeObjects() won't call the overridden methods of App::Part but them of OriginGroupExtension. This is because `grp` and `obj` point to different objects.
So, calling addObjects() or removeObjects()  causes unexpected behaviour.

The same problem occurs if App::Part is accessed by its Python interface.